### ISSUE TITLE
Inconsistent test results

### ISSUE THREAD

**Issue by umaar** [DevExpress/testcafe#1120]
### Are you requesting a feature or reporting a bug?

Bug

### What is the current behavior?

* `1) TypeError: Cannot read property '0' of undefined`

Or I land on this page after a button click

* `http://10.101.115.72:53642/Hkn3vOM8x/about:error`

Or I get this error:

* The element that matches the specified selector is not visible.

### What is the expected behavior?

The test should not fail

#### Provide the test code and the tested page URL (if applicable)

```js
import { Selector } from 'testcafe';

fixture `Example Test`;

async function sleep(ms = 2000) {
	return new Promise(resolve => setTimeout(resolve, ms));
}

test('reproduce bug', async t => {
	await t.navigateTo('http://genomemedicine.biomedcentral.com/');
	await sleep();
	await t.click('.journal-identity__submit');
	await sleep();
});
```

Tested page URL: http://genomemedicine.biomedcentral.com/

### Specify your

* operating system: OS X
* testcafe version: 0.12.0-alpha4 but also tried 0.11.1
* node.js version: v7.0.0

**Comment by AlexanderMoskovkin** [comment]
Hi Umar,

I've reproduced the problem and we are investigating it.

**Comment by churkin** [comment]
Hi Umar, 

I have investigated this problem and found out that obsoleted `framesets` (https://www.w3.org/TR/html/obsolete.html) are used on the problem page. I think this page couldn't ever work. When you succeeded in the tests, did you use `framesets` or you added them and they failed?

**Comment by inikulin** [comment]
@churkin It's odd that we have inconsistent test results if page contains `<frameset>`, can we handle it gracefully? Also, I remember that we've discussed `<frameset>` support, but decided that this obsolete technology doesn't worth the efforts. However, since seems like it's still used in the wild maybe we can support it somehow. I don't remember if we had any technical difficulties with it?

**Comment by umaar** [comment]
Thanks everyone. @churkin ignore what I said about them succeeding previously, it was unrelated tests that were passing. 

Just to be clear, the page I execute the tests on is `http://genomemedicine.biomedcentral.com/` and I am clicking on a hyperlink which takes you to a page which `<framesets>`. I have no desire to test the page with `<framesets>`. Maybe there's a workaround here? Really, the goal of the test is to check there's a working button. The page with the `<frameset>` is not part of our test requirements. It might be enough to `assert` the `href` attribute is correct...that being said, a _good_ test would also click the hyperlink and at least verify the new URL has been navigated to.

**Comment by churkin** [comment]
@umaar @inikulin  Guys, I need some time to assess the cost of `framesets` support

**Comment by umaar** [comment]
@churkin thanks very much. Sorry to be annoying because I agree, `<frameset>` is very old and it would be a shame for you to have to write hacks to support it.

Idea: Don't break so early on just because a `<frameset>` page has been navigated to. If the developer writing a test doesn't need to query the DOM, then nothing needs to break. All I want to do is check the URL and nothing else.

**Comment by inikulin** [comment]
I guess we can provide such limited support by injecting runtime scripts into the first frame of a frameset (and perform HTML instrumentation only for it). Shouldn't be hard to implement. @churkin what do you think?

**Comment by churkin** [comment]
@umaar We have created small fix that will allow you to test problem page https://github.com/DevExpress/testcafe-hammerhead/issues/1009

**Comment by churkin** [comment]
@umaar I've released new 0.12.0-alpha5 version of testcafe, you can try it right now.

**Comment by umaar** [comment]
@churkin @inikulin wow! That's very kind of you. I'll try it out!
